notebooks/HASP/COSLifetimePositions/COSLifetimePositions.ipynb

```diff
@@ -0,0 +1,965 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "acf5a4e7",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2024-04-19T16:53:08.527855Z",
+     "iopub.status.busy": "2024-04-19T16:53:08.527492Z",
+     "iopub.status.idle": "2024-04-19T16:53:11.166493Z",
+     "shell.execute_reply": "2024-04-19T16:53:11.165925Z"
+    }
+   },
+   "source": [
+    "<a id = getdata></a>\n",
+    "### 4.1 Retrieve and Combine the Spectra\n",
+    "The star WD0308-565 (WG 7) is a calibration standard.  Let's use spectra taken at two lifetime positions.  For maximum contrast, we will use LP1 and LP4.  \n",
+    "We begin by setting up our directory structure and querying the MAST database for all COS spectra of the star. This will give us a list of all observations for the program."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d5a4b2a1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the data-download directory.\n",
+    "cos_data_dir = Path(\"./cos_data/\")\n",
+    "\n",
+    "# Define the products directory to hold the output.\n",
+    "cos_products_dir = Path(\"./cos_products/\")\n",
+    "\n",
+    "# If the directories doesn't exist, then create them.\n",
+    "cos_data_dir.mkdir(exist_ok=True)\n",
+    "cos_products_dir.mkdir(exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "77469be9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Conduct the query\n",
+    "wg7_products = Observations.get_product_list(\n",
+    "    Observations.query_criteria(\n",
+    "        instrument_name='COS',\n",
+    "        target_name='WD*0308-565',\n",
+    "        dataproduct_type=\"SPECTRUM\"\n",
+    "    )\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa19d244",
+   "metadata": {},
+   "source": [
+    "This query returns all available data, but we want only the x1d files of spectra taken with the G130M grating.  (This takes a few minutes, so be patient.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b390c561",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "Observations.download_products(\n",
+    "    wg7_products,\n",
+    "    download_dir=str(cos_data_dir),\n",
+    "    filters='G130M',\n",
+    "    productSubGroupDescription=[\"X1D\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "902f9ce3",
+   "metadata": {},
+   "source": [
+    "When we download data using astroquery, it creates a directory called mastDownload/HST. Each dataset goes into a separate subfolder within that directory.  Before running the combination script, we must move all of the x1d files into a single directory. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d6e26ca4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
```

Comment:
i would remove the try/except here. this should work without needing to catch any exception.